src/pal/src/map/virtual.cpp

```diff
@@ -1073,6 +1073,8 @@ VIRTUALCommitMemory(
     nProtect = W32toUnixAccessControl(flProtect);
     vProtect = VIRTUALConvertWinFlags(flProtect);
 
+    StartBoundary = pInformation->startBoundary + runStart * VIRTUAL_PAGE_SIZE;
```

Comment:
I don't think we need it here. If we go to the error below, we were not reserving the memory during commit and so we would not be calling the unmap in the error path. The reason is that if we reserve memory either in this function (which means the IsLocallyReserved is set) or in the caller (which means the MEM_RESERVE is set in the flAllocationType), we have the right amount reserved, otherwise it would be a bug in this code.